### DISCUSSION TITLE
New component - Cookie banner

### DISCUSSION THREAD

**Discussion by a-gonzalo** [sqlpage/SQLPage#233] [Show and tell]
Hi all,

I'm making a component to show directly a cookie banner on the index.sql (or your root path):

-- cookie-notice.handlebars
``` handlebars
<div id="cookie-notice" style="position:fixed; bottom:0; left:0; width:100%; background-color:rgba(0,0,0,0.9); color:white; text-align:center; padding:10px; z-index:1000;">
  <p>{{{markdown banner_md}}}</p>
  <button id="accept-cookie" style="padding: 5px 10px; margin: 10px; background-color: #4CAF50; color: white; border: none; cursor: pointer;">{{accept}}</button>
  <button id="decline-cookie" style="padding: 5px 10px; margin: 10px; background-color: #f44336; color: white; border: none; cursor: pointer;">{{decline}}</button>
</div>
```
-- cookie-notice.js
``` js
document.getElementById('accept-cookie').onclick = function() {
    document.getElementById('cookie-notice').style.display = 'none';
    // TODO
  };
  document.getElementById('decline-cookie').onclick = function() {
    document.getElementById('cookie-notice').style.display = 'none';
    // TODO
  };
```

-- The cookie SQLPage select
``` sql
select 'cookie-notice' as component,
    'This site uses cookies as... cookies. Look the cookie contents [here](https://127.0.0.1)' as banner_md,
    'Accept' as accept,
    'Decline' as decline;
```

-- The final result:

![image](https://github.com/lovasoa/SQLpage/assets/91145535/90ec4bef-3d63-4595-98a7-09d179ba71d5)

**Comment by lovasoa** [comment]
I'm not sure what the audience for such a component would be... Are you hosting third party user tracking scripts on your website? You don't need a cookie banner if you are not doing anything fishy behind the back of your users.

**Reply by a-gonzalo** [reply]
Oh... I didn't know that, sorry.
I just did it because nowadays all the websites had it.

**Reply by lovasoa** [reply]
Because most websites do user tracking. Honest websites don't have it. You won't see a cookie banner on Wikipedia, for instance. If you need cookies for the functionalities your website provides to users, you can use them without a banner. If you use cookies to track your users behind their back, then you need their consent. 

The exact wording is: 

*Exceptions to the
obligation to provide information and offer the right to refuse should be limited to those
situations where the technical storage or access is strictly necessary for the legitimate
purpose of enabling the use of a specific service explicitly requested by the subscriber or
user.”*
